be/src/exec/analytor.cpp

```diff
@@ -523,11 +518,17 @@ Status Analytor::add_chunk(const ChunkPtr& chunk) {
 void Analytor::_append_column(size_t chunk_size, Column* dst_column, ColumnPtr& src_column) {
     if (src_column->only_null()) {
         static_cast<void>(dst_column->append_nulls(chunk_size));
-    } else if (src_column->is_constant()) {
-        auto* const_column = static_cast<ConstColumn*>(src_column.get());
+    } else if (src_column->is_constant() && !dst_column->is_constant()) {
+        // unpack const column, then append it to dst
+        auto* const_column = down_cast<ConstColumn*>(src_column.get());
         const_column->data_column()->assign(chunk_size, 0);
         dst_column->append(*const_column->data_column(), 0, chunk_size);
+    } else if (src_column->is_constant() && dst_column->is_constant() && (!dst_column->empty()) &&
+               (!src_column->empty()) && (src_column->compare_at(0, 0, *dst_column, 1) != 0)) {
+        // i dont't think this will happen
+        DCHECK(0);
```

Comment:
this condition seems cost at runtime. can it be just removed?

Reply:
Or put this inside the DCHECK

Reply:
> Or put this inside the DCHECK
make sense, I will add a DCHECK